test/refunder.ts

```diff
@@ -0,0 +1,299 @@
+			const value = 152;
+
+			let res = await refunder.setMaxGasPrice(value);
+			await res.wait();
+
+			const updatedGasPrice = await refunder.maxGasPrice();
+
+			expect(updatedGasPrice.toString()).to.be.eq(value.toString());
+
+		});
+
+		it("Not Owner should NOT set max gas price", async function() {
+			await expect(refunder.connect(addr1).setMaxGasPrice(191)).to.be.revertedWith(NOT_AN_OWNER);
+		});
+	});
+
+	describe('Whitelist Refundable', () => {
+
+		it("Owner should be able to add refundable", async function() {
+		
+			const randomAddress = greeter.address;
+			const randomFuncIdAsBytes = generateFuncIdAsBytes('setGreeting(string)');
+
+			// address, bytes4
+			const resBefore = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resBefore).to.be.eq(false);
+
+			const res = await refunder.whitelistRefundable(randomAddress, randomFuncIdAsBytes, true);
+			let txReceipt = await res.wait();
+
+			expect(txReceipt.events, 'No events are emitted').to.be.ok;
+			expect(txReceipt.events[0].event, 'Invalid event name').to.be.eq('RefundableUpdate');
+
+			const resAfter = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resAfter).to.be.eq(true);
+		});
+
+		it("Owner should be able to edit refundable", async function() {
+		
+			const randomAddress = greeter.address;
+			const randomFuncIdAsBytes = generateFuncIdAsBytes('setGreeting(string)');
+
+			// address, bytes4
+			const resBefore = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resBefore).to.be.eq(false);
+
+			let res = await refunder.whitelistRefundable(randomAddress, randomFuncIdAsBytes, true);
+			await res.wait();
+
+			const resAfter = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resAfter).to.be.eq(true);
+
+			res = await refunder.whitelistRefundable(randomAddress, randomFuncIdAsBytes, false);
+			await res.wait();
+
+			const resAfterEdit = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resAfterEdit).to.be.eq(false);
+		});
+
+		it("Not owner should NOT be able to add refundable", async function() {
+		
+			const randomAddress = greeter.address;
+			const randomFuncId = ethers.utils.id('setGreeting(string)');
+			const randomFuncIdAsBytes = ethers.utils.arrayify(randomFuncId.substr(0, 10))
+
+			await expect(refunder.connect(addr1).whitelistRefundable(randomAddress, randomFuncIdAsBytes, true)).to.be.revertedWith(NOT_AN_OWNER);
+		});
+
+		it("Not owner should NOT be able to edit refundable", async function() {
+			const randomAddress = greeter.address;
+			const randomFuncIdAsBytes = generateFuncIdAsBytes('setGreeting(string)');
+
+			let res = await refunder.whitelistRefundable(randomAddress, randomFuncIdAsBytes, true);
+			await res.wait();
+
+			await expect(refunder.connect(addr1).whitelistRefundable(randomAddress, randomFuncIdAsBytes, false)).to.be.revertedWith(NOT_AN_OWNER);
+		});
+	});
+
+	describe('Relay and refund', () => {
```

Comment:
We should add a test in which we:
- call `relayAndRefund` for a refundable transaction but the call reverts
Effects:
- The whole TX should revert
- the message sender should not be refunded